stubs/babel/babel/messages/extract.pyi

```diff
@@ -1,30 +1,85 @@
-from collections.abc import Callable
-from typing import Any
+from _typeshed import SupportsRead, SupportsReadline
+from collections.abc import Callable, Collection, Iterable
+from os import PathLike
+from typing import Any, AnyStr, Protocol, overload
+from typing_extensions import TypeAlias, TypedDict
+
+_Keyword: TypeAlias = tuple[int | tuple[int, int] | tuple[int, str], ...]
 
 GROUP_NAME: str
-DEFAULT_KEYWORDS: Any
-DEFAULT_MAPPING: Any
+DEFAULT_KEYWORDS: dict[str, _Keyword]
+DEFAULT_MAPPING: list[tuple[str, str]]
 empty_msgid_warning: str
 
+@overload
+def extract_from_dir(
+    dirname: AnyStr | PathLike[AnyStr],
+    method_map: list[tuple[str, str]] = ...,
+    options_map: dict[str, dict[str, Any]] | None = ...,
+    keywords: dict[str, _Keyword] = ...,
+    comment_tags: Collection[str] = ...,
+    callback: Callable[[AnyStr, str, dict[str, Any]], object] | None = ...,
+    strip_comment_tags: bool = ...,
+    directory_filter: Callable[[str], bool] | None = ...,
+) -> list[tuple[AnyStr, int, str | tuple[str, ...], list[str], str | None]]: ...
```

Comment:
This returns a generator, not  a list. The arguments and return type also applies to more functions and overloads below.